include/tnt/linear/impl/orthogonal_projection_impl.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef TNT_ORTHOGONAL_PROJECTION_IMPL_HPP
+#define TNT_ORTHOGONAL_PROJECTION_IMPL_HPP
+
+#include <tnt/utils/testing.hpp>
+#include <tnt/utils/simd.hpp>
+#include <tnt/linear/orthogonal_projection.hpp>
+
+#define VECTOR_DIM 1
+#define ROW_INDEX 0
+
+namespace tnt {
+
+    namespace detail {
+
+        template<typename DataType>
+
+        struct OptimizedOrthogonalProjection<DataType> {
+
+            static Tensor<DataType> eval(Tensor<DataType> &V1, Tensor<DataType> &V2) {
+                DataType* v = V1.data.data;
+                DataType* s = V2.data.data;
+                int num_rows = V1.shape.axes[ROW_INDEX];
+
+                scalar_multiply(s, (dotProduct(v, s, num_rows) / dotProduct(s, s, num_rows)), num_rows);
```

Comment:
You can just do V2 * dotProduct(...). Scalar multiplication is a method for tensors and should be faster then your version (it uses SIMD)